scripts/views/mapView.js

```diff
@@ -173,6 +187,42 @@
     }
   };
 
+  // If no events of a given type are present in the snapshot, hide its associated legend li
+  mapView.hideAbsentEvents = function hideAbsentEvents(){
+    for(var key in legendIcons){
+      if(legendIcons[key] === false){
+        $(`.${key}-parent`).addClass('hidden');
+      }
+    }
+  };
+
+  // for each townhall event in ele, update legendIcons if key matches
+  mapView.checkForAbsentEvents = function checkForAbsentEvents(ele){
+    if (ele.iconFlag === 'in-person'){
+      legendIcons.inPerson = true;
+    }
```

Comment:
to make this less hard coded could be: 
`legendIcons[ele.iconFlag] = true`
you'd need to change your class names to match the icon flag names though